hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreCompactor.java

```diff
@@ -149,6 +150,8 @@ private void releaseResources() {
    * still need to evaluate the compaction.
    */
   private boolean shouldFlatten() {
+    if (isInterrupted.get())        // if the entire process is interrupted refuse to flatten
+      return false;                 // the compaction also doesn't start when interrupted
     boolean userToFlatten =         // the user configurable option to flatten or not to flatten
         compactingMemStore.getConfiguration().getBoolean(MEMSTORE_COMPACTOR_FLATTENING,
```

Comment:
Looks like there is one more parameter hiding here ... Let's make order in our configuration.